### PR TITLE
feature - GitHub auth 

### PR DESCRIPTION
**Pull request by VedanthB**
adding github auth with firebase

### CODE REVIEW

File: context/FirebaseAuthProvider.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+const AuthContext = createContext({});
+
+const provider = new GithubAuthProvider();
+
+export const AuthContextProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  console.log(user);
```
Comment (prerana1821) [inline context/FirebaseAuthProvider.js:18]: Please remove the logs

File: firebase/firebaseApp.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Import the functions you need from the SDKs you need
+import { initializeApp } from "firebase/app";
+import { getAuth } from "firebase/auth";
+import { getAnalytics } from "firebase/analytics";
+// TODO: Add SDKs for Firebase products that you want to use
+// https://firebase.google.com/docs/web/setup#available-libraries
+
+// Your web app's Firebase configuration
+// For Firebase JS SDK v7.20.0 and later, measurementId is optional
+const firebaseConfig = {
```
Comment (prerana1821) [inline firebase/firebaseApp.js:10]: I guess we should create a `.env` file and add all these values there.
@pranshudobhal your views?

**Review by prerana1821 — CHANGES_REQUESTED** [review]
Great work @VedanthB ! 🔥 🔥 💯

File: firebase/firebaseApp.js (hunk 2 above)
Comment (pranshudobhal) [reply firebase/firebaseApp.js:10]: Yes, we should add them in a `.env.local` file and add it to `.gitignore`.

We should also add the env variables in vercel.

**Pull request merged**